socketio/examples/async_transmitter.rs

```diff
@@ -0,0 +1,116 @@
+use futures_util::FutureExt;
+use rust_socketio::{
+    asynchronous::{Client as SocketIOClient, ClientBuilder as SocketIOClientBuilder},
+    Error as SocketIOError, Payload,
+};
+use serde_json::json;
+use std::sync::{mpsc, Arc};
+use std::time::Duration;
+use tokio::time::sleep;
+
+struct ComplexData {
+    /// There should be many more fields below in real life,
+    /// probaly wrapped in Arc<Mutex<T>> if you're writing a more serious client.
+    data: String,
+}
+
+struct TransmitterClient {
+    receiver: mpsc::Receiver<String>,
+    complex: ComplexData,
+    client: SocketIOClient,
+}
+
+impl TransmitterClient {
+    async fn connect(url: &str) -> Result<Self, SocketIOError> {
+        let (sender, receiver) = mpsc::channel::<String>();
+
+        let client = SocketIOClientBuilder::new(url)
+            .namespace("/admin")
+            .on("test", |payload: Payload, socket: SocketIOClient| {
+                async move {
```

Comment:
Can't we put the `async move` after the closue, e.g.:
`.on("test", |payload: Payload, socket: SocketIOClient| async move {`? That way we have one nesting less.

Reply:
For sure, no problem